src/integrations/retry.ts

```diff
@@ -66,23 +81,29 @@ export async function asyncRetry<T>(
   });
 }
 
-export function retryFetch<T>(
-  url: string,
+export function retryFetch(
+  url: RequestInfo,
   fetchOpts: RequestInit,
   retryOpts?: RetryOptions,
-  parseJson = true,
-): Promise<T> {
+): Promise<Response> {
   return asyncRetry(async () => {
     const res = await fetch(url, fetchOpts);
     if (res.ok) {
-      if (parseJson) {
-        return res.json();
-      }
-      return;
+      return res;
     }
+    const err = new HttpError(url.toString(), res.status, await res.text());
     if (res.status < 500) {
-      throw new AbortError(`request is invalid: ${res.status}`);
+      throw new AbortError(err);
     }
-    throw new Error(`unexpecetd response from feed service: ${res.status}`);
+    throw err;
   }, retryOpts);
 }
+
+export async function retryFetchParse<T>(
```

Comment:
I'd only argue this placement inside `feed` dir is a bit off.

Reply:
It's not in the `feed` dir, it's under integrations, or am I missing something?